classes/submission/reviewAssignment/ReviewAssignmentDAO.inc.php

```diff
@@ -376,15 +375,21 @@ function insertObject($reviewAssignment) {
 		);
 
 		$reviewAssignment->setId($this->getInsertId());
-		return $reviewAssignment->getId();
+
+		// Update review stage status whenever a review assignment is changed
+		if ($result) {
+			$this->updateReviewRoundStatus($reviewAssignment);
+		}
+
+		return $reviewAssignment;
```

Comment:
In this case returning the object makes sense to me because it has modified the object (set an ID based on what was inserted).

I don't have a pattern in mind. But we'll need to return the object or the insert ID to keep the object data current.